lib/screens/music_screen.dart

```diff
@@ -22,6 +23,29 @@ import '../services/finamp_settings_helper.dart';
 import '../services/finamp_user_helper.dart';
 import '../services/jellyfin_api_helper.dart';
 
+void postLaunchHook(WidgetRef ref) async {
+  final downloadsService = GetIt.instance<DownloadsService>();
+  final queueService = GetIt.instance<QueueService>();
+
+  // make sure playlist info is downloaded for users upgrading from older versions and new installations AFTER logging in and selecting their libraries/views
+  if (!FinampSettingsHelper.finampSettings.hasDownloadedPlaylistInfo) {
+    // check if metadata already downloaded to avoid possible duplicates
+    DownloadItemStatus? status = downloadsService.getStatus(
+        DownloadStub.fromFinampCollection(
+            FinampCollection(type: FinampCollectionType.allPlaylistsMetadata)),
+        null);
+    if (status.isRequired == false) {
+      await downloadsService.addDefaultPlaylistInfoDownload();
```

Comment:
I don't really expect there to be errors. but it might make sense to put a catchError on this just in case?

Reply:
Never hurts. Done!